jvb/src/main/kotlin/org/jitsi/videobridge/message/BridgeChannelMessage.kt

```diff
@@ -398,20 +402,40 @@ class SenderVideoConstraintsMessage(val videoConstraints: VideoConstraints) : Br
     }
 }
 
+/**
+ * A message sent from the bridge to a client (sender), indicating constraints for the sender's video stream.
+ */
+class SenderSourceConstraintsMessage(
+    val sourceName: String,
+    val maxHeight: Int
+) : BridgeChannelMessage(TYPE) {
+
+    /**
+     * Serialize manually because it's faster than Jackson.
+     */
+    override fun createJson(): String =
+        """{"colibriClass":"$TYPE", "sourceName":"$sourceName", "maxHeight":$maxHeight}"""
+
+    companion object {
+        const val TYPE = "SenderSourceConstraints"
+    }
+}
+
 /**
  * A message sent from one bridge to another (via Octo) indicating that the first bridge wishes to receive video streams
  * from the specified endpoint with the specified constraints.
  */
 class AddReceiverMessage(
     val bridgeId: String,
-    val endpointId: String,
+    val endpointId: String?, // Used in single stream per endpoint mode and wil be removed
+    val sourceName: String?, // Used in the multi-stream mode
     val videoConstraints: VideoConstraints
 ) : BridgeChannelMessage(TYPE) {
     /**
      * Serialize manually because it's faster than Jackson.
      */
     override fun createJson(): String =
-        """{"colibriClass":"$TYPE","bridgeId":"$bridgeId","endpointId":"$endpointId",""" +
+        """{"colibriClass":"$TYPE","bridgeId":"$bridgeId","endpointId":"$endpointId","sourceName":"$sourceName",""" +
```

Comment:
The serialization is broken now that some of the values are nullable

Reply:
broken in what way? that it puts `null` there? I can make it skip that. I'm asking because I was testing Octo with 3 JVBs and saw no exception about serialization

Reply:
You can play with the perf test in `BridgeChannelMessageTest`, but I'm not too worried about the performance -- it's not one of the messages that are sent often